foreman/data_refinery_foreman/foreman/management/commands/rerun_salmon_old_samples.py

```diff
@@ -0,0 +1,108 @@
+""" This management command will search for experiments where samples haven't been processed and
+that multiple versions of salmon have been used on the samples.
+"""
+
+import random
+import sys
+import time
+from typing import Dict, List
+
+from django.core.management.base import BaseCommand
+from nomad import Nomad
+from django.db.models import OuterRef, Subquery
+from django.db.models.expressions import Q
+
+from data_refinery_common.models import (
+    DownloaderJob,
+    DownloaderJobOriginalFileAssociation,
+    Experiment,
+    ExperimentOrganismAssociation,
+    ExperimentSampleAssociation,
+    Organism,
+    OriginalFile,
+    ProcessorJob,
+    ProcessorJobOriginalFileAssociation,
+    Sample,
+    ComputationalResult,
+)
+from data_refinery_common.job_lookup import ProcessorEnum, ProcessorPipeline, Downloaders
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.message_queue import send_job
+from data_refinery_common.rna_seq import get_quant_results_for_experiment
+from data_refinery_common.utils import get_env_variable, get_active_volumes
+from data_refinery_common.job_management import create_processor_job_for_original_files
+
+logger = get_and_configure_logger(__name__)
+
+def update_salmon_versions(experiment: Experiment):
+    quant_results = get_quant_results_for_experiment(experiment)
+    salmon_versions = list(quant_results.order_by('-organism_index__created_at')\
+                                   .values_list('organism_index__salmon_version', flat=True)\
+                                   .distinct())
+    
+    if len(salmon_versions) <= 1:
+        # only apply this command on experiments that have more than one salmon version applied on their samples
+        return
+
+    latest_salmon_version = salmon_versions[0]
+
+    # find the samples that were not processed with `latest_salmon_version` and trigger new processor jobs for them
+    newest_computational_results = ComputationalResult.objects.all()\
+        .filter(
+            samples=OuterRef('id'),
+            processor__name=ProcessorEnum.SALMON_QUANT.value['name']
+        )\
+        .order_by('-created_at')
+    
+    samples = experiment.samples.all().annotate(
+            salmon_version=Subquery(newest_computational_results.values('organism_index__salmon_version')[:1])
```

Comment:
Can you test that this is set to None if the sample has no computational results? (i.e., not an error)